### PR TITLE
Statistics ilai

### PR DESCRIPTION
**Pull request by ilaiazulay**

### CODE REVIEW

File: controller/stats.controller.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ const ServerError = require("../errors/internal.errors");
 const ExperimentRepository = require("../repositories/experiment.repository");
 const UserRepository = require("../repositories/user.repository");
 const {mongoose} = require("mongoose");
+const Experiment = require("../models/Experiment.model");
```
Comment (peerfikhman) [inline controller/stats.controller.js:7]: same as before it is advised to creat a method in repository and only use the repository

File: controller/stats.controller.js [hunk 2]
```diff
@@ -69,7 +71,26 @@ const getUsersStats = async (req, res) => {
     }
 }
 
+const getTestsPerMonth = async (req, res) => {
+
+    const accountID = req.params.accountId;
+    if (!mongoose.isValidObjectId(accountID)) throw new ValidationError.MissingPropertyError("account ID");
```
Comment (peerfikhman) [inline controller/stats.controller.js:77]: Great Validation!

File: controller/stats.controller.js [hunk 3]
```diff
@@ -69,7 +71,26 @@ const getUsersStats = async (req, res) => {
...
+    Experiment.aggregate([
```
Comment (peerfikhman) [inline controller/stats.controller.js:78]: Very nice logic in this function and a very elegant way of coding
though this should not be here, this should be in the Experiment Repository as mentioned above

File: Service/cron.job.js [hunk 4]
```diff
@@ -16,6 +17,15 @@ const experimentStatusUpdate = cron.schedule("* 0 * * *", async () => {
     });
 })
 
+cron.schedule("0 0 1 * *", async () => {
+  try {
+    await Experiment.updateMany({}, { $set: { monthlyCallCount: 0 } });
+    console.log("Monthly call count reset successful");
```
Comment (peerfikhman) [inline Service/cron.job.js:23]: although you are correct that we should log this data, console log is no the way of doing this

File: Service/cron.job.js [hunk 5]
```diff
@@ -16,6 +17,15 @@ const experimentStatusUpdate = cron.schedule("* 0 * * *", async () => {
...
+  } catch (err) {
+    console.error(err);
```
Comment (peerfikhman) [inline Service/cron.job.js:25]: errors should be handled with error handlers

File: router/stats.router.js [hunk 6]
```diff
@@ -4,6 +4,7 @@ const statsRouter = new Router();
 
 statsRouter.get("/userVariant/:id", statsController.getUsersStats)
 statsRouter.get('/:id',statsController.getStatistics);
+statsRouter.get('/testsPerMonth/:accountId',statsController.getTestsPerMonth);
```
Comment (peerfikhman) [inline router/stats.router.js:7]: good job on the Router very nice looking

**Review by peerfikhman — CHANGES_REQUESTED** [review]
Hey @ilaiazulay,
I reviewed your code, very good job!
There are  though some conventions that we must follow such as working on the data base through the Repositories and not directly on the model, I very much advise to fix it before merge.

Regarding the crone job, please checkout my comments,
I advise to consult with @AtomicManic and @sufka861 about the way of doing this,
I haven't put my mind into the correct solution but I think that saving each month's statistics in the data base is much more advised then doing a cron job every month and resetting the data resulting in loosing that data.

File: Service/cron.job.js [hunk 7]
```diff
@@ -16,6 +17,15 @@ const experimentStatusUpdate = cron.schedule("* 0 * * *", async () => {
     });
 })
 
+cron.schedule("0 0 1 * *", async () => {
```
[the remark this answers is not in the document]
Comment (AtomicManic) [reply Service/cron.job.js:20]: Read some of the comments, what Peer is suggesting is correct. Better make a collection of data logs and take it from there per call

**Review by peerfikhman — APPROVED** [review]
Good job on the fixes!
Merge approved!

**Pull request merged**